features/admin.applications.v1/components/settings/shared-access.tsx

```diff
@@ -53,15 +61,82 @@ export const SharedAccess: FunctionComponent<SharedAccessPropsInterface> = (
 ): ReactElement => {
 
     const { application, onUpdate, readOnly } = props;
+    const [ sharingState, setSharingState ] = useState<OperationStatus>(OperationStatus.IDLE);
+    const [ sharingOperationId, setSharingOperationId ] = useState<string>();
+    const disabledFeatures: string[] = useSelector((state: AppState) =>
+        state?.config?.ui?.features?.applications?.disabledFeatures);
+    const isApplicationShareOperationStatusEnabled: boolean =
+        disabledFeatures?.includes("applications.sharedAccess.status") ?? false;
+    const applicationShareStatusPollInterval: number = useSelector((state: AppState) =>
+        state?.config?.ui?.features?.applications?.applicationShareStatusPollInterval);
+    const [ sharingOperationSummary, setSharingOperationSummary ] = useState<OperationStatusSummary>();
+    const dispatch: Dispatch = useDispatch();
+    const { t } = useTranslation();
+
+    const { status, startPolling } = useOperationStatusPoller({
+        enabled: isApplicationShareOperationStatusEnabled,
+        onCompleted: (finalStatus: OperationStatus) => {
+            setSharingState(finalStatus);
+            if (status === OperationStatus.FAILED) {
+                dispatch(addAlert({
```

Comment:
instead of three dispatch statements, let's define one variable to store the status and use that to construct the i18n strings for message and description.

eg:

```js
const statusToi18nKeyMap = {
   [OperationStatus.SUCCESS]: "success",
   [OperationStatus.FAILURE]: "failure",
   // rest
}

dispatch(addAlert({
       description: t("applications:edit.sections.shareApplication.completedSharingNotification."
                        + statusToi18nKeyMap[status] + ".description"),
       level: AlertLevels.SUCCESS,
       message: t("applications:edit.sections.shareApplication.completedSharingNotification."
                        + statusToi18nKeyMap[status]  + ".message")
}));
```